packages/next/src/build/utils.ts

```diff
@@ -1953,7 +1953,7 @@ if (!process.env.NEXT_MANUAL_SIG_HANDLE) {
 }
 
 const currentPort = parseInt(process.env.PORT, 10) || 3000
-const hostname = process.env.HOSTNAME || 'localhost'
+const hostname = process.env.HOSTNAME || '127.0.0.1'
```

Comment:
I think on some operating systems `127.0.0.1` is not exposed outside the current device, is that the desired behaviour?

`0.0.0.0` would bind to all network interfaces (available in the local network)

Reply:
I think somewhere higher up in the tree we aren't mapping localhost -> `127.0.0.1` correctly

Reply:
I believe this https://github.com/vercel/next.js/pull/52283/commits/7aa687919d65e2805e23b5f746e59ead0a83e6d3 is where want to fix it to match `next start` behavior.

Reply:
x-ref: https://github.com/vercel/next.js/blob/31dee6f38add2032d217967cbbc61567244804ce/packages/next/src/server/lib/start-server.ts#L268